### ISSUE TITLE
Update Python packaging  according to PEP 517, 518 

### ISSUE THREAD

**Issue by jruebsam** [OpenSimulationInterface/open-simulation-interface#571]
#### Build-independent format for source tree 

This would be a desired feature somewhere in the near future. 
Currently  a lot of python projects shift towards more advanced build tools like e.g. 

https://python-poetry.org/
https://pipenv.pypa.io/en/latest/
https://pdm.fming.dev/

These tools require a build-independent format (which is often not relying on setuptools / distutils) 
furthermore this tools can satisfy deterministic builds easily. 

The definition of this format is officially defined by PEP 517 and PEP 518, e.g.:

https://www.python.org/dev/peps/pep-0517/
https://www.python.org/dev/peps/pep-0518/

So basically what this means would be at some point to create a `pyproject.toml` which defines 
all mandatory build / version specifications.

Since this project doesn't rely on a lot of dependencies, I think for now this is not an important feature,
but this should be addressed at some point in the future

Cheers

**Comment by pmai** [comment]
Implemented with split out of osi-python.
